### PR TITLE
schema(cdc): fix DML construct error caused by 'rename tables' DDL

### LINKED ISSUE #5059: Error DDL/DML schedule， which cause error DML rewrite
### What did you do?

1. Setup upstream and downstream cluster
2. Test tidb-mysql-test `bigint` test case, need change all `drop table` to rename `table`
3.  check data consistency

### What did you expect to see?

changefeed work normal

### What did you see instead?

Error DML rewrite and cause changefeed stop by error.
It may cause by incorrect `rename table` implemement.

sql sequence:
```
create table t1 (
 value64  bigint unsigned  not null,
 value32  integer          not null,
 primary key(value64, value32)
);

create table t2 (
 value64  bigint unsigned  not null,
 value32  integer          not null,
 primary key(value64, value32)
);

insert into t1 values(17156792991891826145, 1);
insert into t1 values( 9223372036854775807, 2);
insert into t2 values(17156792991891826145, 3);
insert into t2 values( 9223372036854775807, 4);

select * from t1;
select * from t2;

select * from t1, t2 where t1.value64=17156792991891826145 and
t2.value64=17156792991891826145;
select * from t1, t2 where t1.value64=17156792991891826145 and
t2.value64=t1.value64;

select * from t1, t2 where t1.value64= 9223372036854775807 and
t2.value64=9223372036854775807;
select * from t1, t2 where t1.value64= 9223372036854775807 and
t2.value64=t1.value64;

rename table t1 to t1_7, t2 to t2_7;
```

log:
```
[2022/03/29 06:33:43.930 +00:00] [DEBUG] [mysql.go:768] ["prepare DMLs"] [rows="[{\"start-ts\":432153712015441934,\"commit-ts\":432153712015441938,\"row-id\":1,\"table\":{\"db-name\":\"bigint\",\"tbl-name\":\"t2_7\",\"tbl-id\":84,\"is-partition\":false},\"column-infos\":[{\"ID\":1,\"IsPKHandle\":false,\"VirtualGenCol\":false,\"Ft\":{\"Tp\":8,\"Flag\":4131,\"Flen\":20,\"Decimal\":0,\"Charset\":\"binary\",\"Collate\":\"binary\",\"Elems\":null}},{\"ID\":2,\"IsPKHandle\":false,\"VirtualGenCol\":false,\"Ft\":{\"Tp\":3,\"Flag\":4099,\"Flen\":11,\"Decimal\":0,\"Charset\":\"binary\",\"Collate\":\"binary\",\"Elems\":null}}],\"table-info-version\":432153712028549131,\"replica-id\":0,\"columns\":[{\"name\":\"value64\",\"type\":8,\"charset\":\"binary\",\"flag\":187,\"value\":17156792991891826145},{\"name\":\"value32\",\"type\":3,\"charset\":\"binary\",\"flag\":59,\"value\":3}],\"pre-columns\":null}]"] [sqls="[\"REPLACE INTO `bigint`.`t2_7`(`value64`,`value32`) VALUES (?,?)\"]"] [values="[[17156792991891826145,3]]"]
[2022/03/29 06:33:43.932 +00:00] [DEBUG] [mysql.go:614] ["exec row"] [sql="REPLACE INTO `bigint`.`t2_7`(`value64`,`value32`) VALUES (?,?)"] [args="[17156792991891826145,3]"]
[2022/03/29 06:33:44.293 +00:00] [ERROR] [feed_state_manager.go:344] ["processor report an error"] [changefeed=tidb-mysql-test] [captureID=87ffe707-65f7-4fae-a0c8-67c545fbe9bd] [error="{\"addr\":\"127.0        .0.1:8300\",\"code\":\"CDC:ErrProcessorUnknown\",\"message\":\"[CDC:ErrMySQLTxnError]MySQL txn error: Error 1146: Table 'bigint.t2_7' doesn't exist\"}"]
```
[cdc 5.log](https://github.com/pingcap/tiflow/files/8369046/cdc.5.log)

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
Release Version: v6.1.0-alpha-77-gcd0d6b85a
Edition: Community
Git Commit Hash: cd0d6b85a92ad2ee5da735563603b1588c1d1463
Git Branch: master
UTC Build Time: 2022-03-28 11:12:19
GoVersion: go1.18
Race Enabled: false

```

Upstream TiKV version (execute `tikv-server --version`):

```console
Release Version:   6.0.0-alpha
Edition:           Community
Git Commit Hash:   4f3e720e1ec2e4e2f8d1ce2c0151a0879b01b885
Git Commit Branch: master
UTC Build Time:    2022-03-28 11:09:04
Rust Version:      rustc 1.60.0-nightly (1e12aef3f 2022-02-13)
Enable Features:   jemalloc mem-profiling portable sse test-engines-rocksdb cloud-aws cloud-gcp cloud-azure
Profile:           dist_release
```

TiCDC version (execute `cdc version`):

```console
Release Version: v6.0.0-master-dirty
```

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve?

Issue Number:  close #5059

### What is changed and how it works?
Don't return directly for 'rename tables' DDL logic in  schema_storge.go:handleDDL,
ref: https://github.com/pingcap/tiflow/blob/fffe5a1049c8e14caa344cc8bece5648a22ca96c/cdc/entry/schema_storage.go#L606

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update key monitor metrics in both TiCDC document and official document

### Release note 

```release-note
fix DML construct error issue caused by `rename tables` DDL.
```

### CODE REVIEW

File: cdc/processor/processor.go [hunk 1]
```diff
@@ -935,6 +935,9 @@ func (p *processor) createTablePipelineImpl(ctx cdcContext.Context, tableID mode
 		p.sendError(err)
 		return nil
 	})
+
+	// TODO: using GetLastSnapshot here would be confused and get the wrong table name
```
Comment (overvenus) [inline cdc/processor/processor.go:939]:
```suggestion
	// FIXME: using GetLastSnapshot here would be confused and get the wrong table name
```

Should it be FIXME?

**Review by hicqu — APPROVED** [review]

File: tests/integration_tests/rename_tables/data/test.sql [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ value32  integer          not null,
+ primary key(value64, value32)
+);
+
+insert into t1 values(17156792991891826145, 1);
+insert into t1 values( 9223372036854775807, 2);
+insert into t2 values(17156792991891826145, 3);
+insert into t2 values( 9223372036854775807, 4);
+
+rename table t1 to t1_7, t2 to t2_7;
```
Comment (overvenus) [inline tests/integration_tests/rename_tables/data/test.sql:44]: Can you insert some row to `t1_7` and `t2_7` after renaming?

**Review by overvenus — APPROVED** [review]
Awesome!

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**